src/sage/matroids/matroid.pyx

```diff
@@ -2882,32 +2969,90 @@ cdef class Matroid(SageObject):
         r"""
         Return the `f`-vector of the matroid.
 
-        The `f`-*vector* is a vector `(f_0, ..., f_r)`, where `f_i` is the
-        number of flats of rank `i`, and `r` is the rank of the matroid.
+        The `f`-*vector* is a vector `(f_0, \ldots, f_r)`, where `f_i` is the
+        number of independent sets of rank `i`, and `r` is the rank of the
+        matroid.
+
+        OUTPUT: a list of integers
 
-        OUTPUT:
+        EXAMPLES::
+
+            sage: M = matroids.catalog.Vamos()
+            sage: M.f_vector()
+            [1, 8, 28, 56, 65]
+
+        TESTS::
+
+            sage: for M in matroids.AllMatroids(5):                                     # optional - matroid_database
+            ....:     assert M.f_vector() == SimplicialComplex(M.bases()).f_vector()
+        """
+        cdef list f = []
+        cdef int i, sum
+        for i in range(self.full_rank() + 1):
+            sum = 0
+            for _ in self.independent_r_sets_iterator(i):
+                sum += 1
+            f.append(sum)
```

Comment:
```suggestion
        cdef int i, s
        for i in range(self.full_rank() + 1):
            s = 0
            for _ in self.independent_r_sets_iterator(i):
                sum += 1
            f.append(ZZ(sum))
```
Two little things here: let's not use Python special function names and we should try to make sure `int`'s don't leak out.